drop-transfer/src/file/mod.rs

```diff
@@ -337,9 +340,15 @@ where
         reader.consume(n);
 
         total_n += n as u64;
-        if let Some(progress_cb) = progress_cb.as_mut() {
-            progress_cb(total_n).await;
+        from_last_event += n as u64;
+
+        if let (Some(progress_cb), Some(granularity)) = (progress_cb.as_mut(), event_granularity) {
+            if from_last_event >= granularity {
+                progress_cb(total_n).await;
+                from_last_event -= granularity;
+            }
         }
```

Comment:
This is minor since the logic is fine, but I must ask why you didn't implement the check for the event in the callback `progress_cb`? This would save you from pushing `event_granularity` down the call stack.
As I said the logic is fine and you can ignore this comment.

Reply:
Well, I had a hard time trying to store info in the progress_cb :sweat_smile:  and to avoid logic duplication there are 2 places where the callback is declared.